### PR TITLE
[Fluid] [Compressible] Updated shock capturing entropy based files

### PR DESCRIPTION
**Pull request by Marco1410**
**📝 Description**
The minimal necessary changes to add the gas constant to the formulation. The constant is implicit in the temperature equation used for the authors.

**🆕 Changelog**
- Added the gas constant to the formulation.

### CODE REVIEW

File: applications/FluidDynamicsApplication/custom_processes/shock_capturing_entropy_viscosity_process.cpp [hunk 1]
```diff
@@ -337,7 +339,8 @@ std::tuple<ShockCapturingEntropyViscosityProcess::TotalDerivativeUtil, ShockCapt
 ShockCapturingEntropyViscosityProcess::BuildTotalDerivativeUtils(
     const Element& rElement,
     const double DeltaTime,
-    const double HeatCapacityRatio)
+    const double gamma,
+    const double c_v)
```
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/custom_processes/shock_capturing_entropy_viscosity_process.cpp]:
```suggestion
    const double HeatCapacityRatio,
    const double SpecificHeatCV)
```
So we use the same argument naming that you used below. Note that these also match the style guide.

File: applications/FluidDynamicsApplication/custom_processes/shock_capturing_entropy_viscosity_process.h [hunk 2]
```diff
@@ -287,7 +287,8 @@ class KRATOS_API(FLUID_DYNAMICS_APPLICATION) ShockCapturingEntropyViscosityProce
     static InfNormData ComputeElementalInfNormData(
         const Element& rElement,
         const double DeltaTime,
-        const double HeatCapacityRatio);
+        const double HeatCapacityRatio,
+        const double SpecificHeatCV);
```
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/custom_processes/shock_capturing_entropy_viscosity_process.h:291]: Shouldn't `SpecificHeatCV` be heat capacity?

File: applications/FluidDynamicsApplication/custom_processes/shock_capturing_entropy_viscosity_process.cpp (hunk 1 above)
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/custom_processes/shock_capturing_entropy_viscosity_process.cpp:343]: You can also use `HeatCapacityRatio` for gamma as you did below.

File: applications/FluidDynamicsApplication/custom_processes/shock_capturing_entropy_viscosity_process.h [hunk 3]
```diff
@@ -298,7 +299,8 @@ class KRATOS_API(FLUID_DYNAMICS_APPLICATION) ShockCapturingEntropyViscosityProce
     static std::tuple<TotalDerivativeUtil, TotalDerivativeUtil, Vector> BuildTotalDerivativeUtils(
         const Element& rElement,
         const double DeltaTime,
-        const double HeatCapacityRatio);
+        const double HeatCapacityRatio,
+        const double c_v);
```
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/custom_processes/shock_capturing_entropy_viscosity_process.h]:
```suggestion
        const double SpecificHeatCV);
```
Be consistent among similar functions, this eases going through the code later on.

File: applications/FluidDynamicsApplication/custom_processes/shock_capturing_entropy_viscosity_process.cpp [hunk 4]
```diff
@@ -355,7 +358,7 @@ ShockCapturingEntropyViscosityProcess::BuildTotalDerivativeUtils(
         const auto& velocity = r_node.FastGetSolutionStepValue(VELOCITY);
         const double temperature = r_node.FastGetSolutionStepValue(TEMPERATURE);
 
-        total_velocities[i] = norm_2(velocity) + std::sqrt(HeatCapacityRatio * temperature);
+        total_velocities[i] = norm_2(velocity) + std::sqrt(gamma * (gamma - 1) * c_v * temperature);
```
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/custom_processes/shock_capturing_entropy_viscosity_process.cpp]:
```suggestion
        total_velocities[i] = norm_2(velocity) + std::sqrt(HeatCapacityRatio * (HeatCapacityRatio - 1) * SpecificHeatCV * temperature); 
```

**Review by rubenzorrilla — COMMENTED** [review]
Minor observations about the naming of the method arguments. Aside of these, this should be ready.

**Comment by Marco1410** [comment]
I committed the changes. Thanks for the suggestions.

**Review by rubenzorrilla — APPROVED** [review]
Should be ready now. Congratulations on your first PR. :wink: :rocket:

**Pull request merged**